apps/calendar/test/unit/views/settings_test.js

```diff
@@ -344,8 +349,23 @@ suiteGroup('views/settings', function() {
       calledWith = arguments;
     };
 
+    assert.notOk(subject.syncProgress.getAttribute('data-l10n-id'));
+    assert.notOk(subject.syncProgress.classList.contains('syncing'));
+
     triggerEvent(subject.syncButton, 'click');
     assert.ok(calledWith);
+
+    controller.on('syncStart', function () {
+      assert.ok(subject.syncProgress.classList.contains('syncing'));
+      assert.equal(subject.syncProgress.getAttribute('data-l10n-id'),
+        'sync-progress-syncing');
```

Comment:
@millermedeiros thanks for the explanation. I made the changes you recommended as it's probably enough in this case to test the actions of the events. The only change I made was to call app.syncController directly in the test blocks as app does not seem to be in scope for the suite.